civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -1723,6 +1731,32 @@ function civihr_employee_portal_menu() {
   return $items;
 }
 
+/**
+ * Callback function for 'emergency_contacts/{id}/delete
+ *
+ * @param string $id
+ *   The emergency contact ID
+ */
+function civihr_employee_portal_emergency_contact_delete($id) {
+  $service = new EmergencyContactDeletionNotifier();
+  $result = ['status' => 200];
+
+  try {
+    civicrm_api3('Contact', 'deleteemergencycontact', ['id' => $id]);
+    $service->notifyDeletion($id);
```

Comment:
The `notifiyDeletion()` method loads the emergency contact with the given ID, but at this point, it was already deleted. Won't this be a problem?

Reply:
Yep, my fault - I should have checked this but it was silently failing and showing an empty contact name in the email. I've fixed this now by fetching the contact before deletion and then passing it to the notifier